bindings/wallet-core/src/wallet.rs

```diff
@@ -173,4 +180,33 @@ impl Wallet {
     pub fn set_state(&mut self, value: Value, counter: u32) {
         self.account.update_state(value, counter)
     }
+
+    /// Cast a vote
+    pub fn vote(
+        &mut self,
+        settings: Settings,
+        plan: VotePlanId,
+        proposal_index: u8,
+        choice: Choice,
+    ) -> Result<Transaction<VoteCast>, Error> {
```

Comment:
So this is going to the right direction. However this is missing too many parameters.

Missing:

* vote options (`vote::Options`), so we can check the choice is within the options of the proposals, this is important as we need to future proof the API as much as possible;
* the payload type (`vote::PayloadType`): this is important also to future proof the API. Right now we know this is going to be `Public` because that is the only available option, however this will change, so we need to have the right data here.

@eugene-babichenko , I know you were going to work on this too. Can you pick up from there and work with @enzoc4 to add the necessary functionality. Ideally I think we need to construct a `Proposal` object that we can build from the REST API (or retrieve from the blockchain, for example, in the first iterations, the proposals are all going to be in the block0, so we should be able to have them with the `Settings`).

Reply:
Just a note, the REST API provides all the information to construct the `Proposal`, so I don't think there is a need to fetch anything from block0, unless we want to doublecheck.